contracts/Pool.sol

```diff
@@ -28,3 +28,84 @@ contract Pool {
         receiver.transfer(value);
     }
 }
+
+struct ReceiverWeight {
+    address next;
+    uint32 weight;
+}
+
+/// @notice Helper methods for receiver weights list.
+/// The list works optimally if after applying a series of changes it's iterated over.
+/// The list uses 1 word of storage per receiver with a non-zero weight.
+library ReceiverWeightsImpl {
+    address internal constant ADDR_ROOT = address(0);
+    address internal constant ADDR_UNINITIALIZED = address(0);
+    address internal constant ADDR_END = address(1);
+
+    function init(mapping(address => ReceiverWeight) storage self) internal {
+        self[ADDR_ROOT].next = ADDR_END;
+    }
+
+    /// @notice Return the next non-zero receiver weight and its address.
+    /// Removes all the zeroed items found between the current and the next receivers.
+    /// Iterating over the whole list removes all the zeroed items.
+    /// @param current The previously returned receiver address or zero to start iterating
+    /// @return next The next receiver address
+    /// @return weight The next receiver weight, zero if the end of the list was reached
+    function nextWeight(
+        mapping(address => ReceiverWeight) storage self,
+        address current
+    ) internal returns (address next, uint32 weight) {
+        ReceiverWeight storage currentReceiverWeight = self[current];
+        next = currentReceiverWeight.next;
+        weight = 0;
+        if (next != ADDR_END) {
+            ReceiverWeight storage nextReceiverWeight = self[next];
+            weight = nextReceiverWeight.weight;
+            // remove elements being zero
+            if (weight == 0) {
+                do {
+                    next = nextReceiverWeight.next;
+                    // delete nextReceiverWeight to release its memory
+                    nextReceiverWeight.next = ADDR_UNINITIALIZED;
```

Comment:
Is there a better solution? `delete nextReceiverWeigth;` fails to compile, because Solidity thinks that I want to clear the pointer and make it invalid. I don't know how to tell it that I want to delete the pointee.

Reply:
Bypassed